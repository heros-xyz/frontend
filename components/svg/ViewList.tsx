import { Icon, IconProps } from "@chakra-ui/react";

const ViewList: React.FC<IconProps> = (props) => {
  return (
    <Icon
      width="100%"
      height="100%"
      viewBox="0 0 18 14"
      fill="none"
      xmlns="http://www.w3.org/2000/svg"
      {...props}
    >
      <path
        d="M1 1H17M1 5H17M1 9H17M1 13H17"
        stroke="#313F4C"
        strokeWidth="2"
        strokeLinecap="round"
        strokeLinejoin="round"
      />
    </Icon>
  );
};
export default ViewList;
