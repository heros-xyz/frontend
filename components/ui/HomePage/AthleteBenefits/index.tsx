import { Box, Button, Flex, Text } from "@chakra-ui/react";
import React from "react";

interface IAthleteBenefitsFeature {
  title: string;
  content: string;
  icon: React.ReactElement;
}

interface IAthleteBenefits {
  title: string;
  description: string;
  features: Array<IAthleteBenefitsFeature>;
}

const AthleteBenefits: React.FC<IAthleteBenefits> = ({
  title,
  description,
  features,
}) => {
  return (
    <Box bg="primary" color="white" px={5} py={16}>
      <Box mb={12}>
        <Text textTransform={"uppercase"} fontWeight="bold" fontSize="md">
          {title}
        </Text>
        <Text
          mt={2}
          fontSize="2xl"
          fontWeight="extrabold"
          lineHeight={8}
          color="accent.3"
        >
          {description}
        </Text>
      </Box>
      <Box mb={12}>
        {features.map((benefit) => {
          return (
            <Flex direction="row" key={benefit.title} mb={10}>
              <Box
                bgColor="secondary"
                minWidth="48px"
                height="48px"
                borderRadius="md"
              >
                <Flex alignItems="center" justify="center" w="100%" h="100%">
                  <Box w="24px" h="24px">
                    {benefit.icon}
                  </Box>
                </Flex>
              </Box>
              <Flex direction="column" ml={4} alignItems="start">
                <Text
                  textTransform={"capitalize"}
                  fontWeight="bold"
                  fontSize="xl"
                  lineHeight="140%"
                  mb={2}
                >
                  {benefit.title}
                </Text>
                <Text
                  fontWeight="normal"
                  fontSize="md"
                  lineHeight="140%"
                  color="grey.100"
                >
                  {benefit.content}
                </Text>
              </Flex>
            </Flex>
          );
        })}
      </Box>
      <Box
        display={"flex"}
        justifyContent="center"
        textTransform={"uppercase"}
        color="primary"
      >
        <Button height={12} width="full" bg="secondary">
          become an athlete
        </Button>
      </Box>
    </Box>
  );
};
export default AthleteBenefits;
