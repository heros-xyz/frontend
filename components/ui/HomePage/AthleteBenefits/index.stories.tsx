import type { Meta, StoryObj } from "@storybook/react";
import React from "react";
import ViewList from "@components/svg/ViewList";
import ViewBoard from "@components/svg/ViewBoard";
import Calendar from "@components/svg/Calendar";
import Users from "@components/svg/Users";
import AthleteBenefits from "./index";

const meta: Meta<typeof AthleteBenefits> = {
  title: "HomePage/AthleteBenefits",
  component: AthleteBenefits,
  tags: ["autodocs"],
  parameters: {
    layout: "fullscreen",
  },
};

export default meta;
type Story = StoryObj<typeof AthleteBenefits>;
const AthleteBenefitsMockData = [
  {
    title: "Benefit No.1",
    content:
      "Nunc a, lacinia sed risus neque, arcu, rhoncus. Id mauris justo facilisis aliquam platea vestibulum condimentum morbi.",
    icon: <ViewList />,
  },
  {
    title: "Benefit No.2",
    content:
      "Purus lobortis volutpat posuere id integer nunc tellus. Non mauris malesuada feugiat massa mi pellentesque cum est. Pharetra a varius urna rhoncus, tempor rutrum.",
    icon: <ViewBoard />,
  },
  {
    title: "Benefit No.3",
    content:
      "Purus lobortis volutpat posuere id integer nunc tellus. Non mauris malesuada feugiat massa mi pellentesque cum est. Pharetra a varius urna rhoncus, tempor rutrum.",
    icon: <Calendar />,
  },
  {
    title: "Benefit No.4",
    content:
      "Tincidunt sollicitudin interdum nunc sit risus at bibendum vitae. Urna, quam ut sit justo non, consectetur et varius.",
    icon: <Users />,
  },
];
export const AthleteBenefitsTemplate: Story = {
  args: {
    title: "athletesâ€™ benefits",
    description:
      "Get yourself an opportunity to earn money and interact with your fans at the same time!",
    features: AthleteBenefitsMockData,
  },
};
