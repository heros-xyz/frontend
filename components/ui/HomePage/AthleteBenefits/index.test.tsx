import { render } from "@testing-library/react";
import React from "react";
import Calendar from "@/components/svg/Calendar";
import Users from "@/components/svg/Users";
import ViewBoard from "@/components/svg/ViewBoard";
import ViewList from "@/components/svg/ViewList";
import AthleteBenefits from "./index";

it("renders athlete benefits unchanged", () => {
  const AthleteBenefitsMockData = [
    {
      title: "Benefit No.1",
      content:
        "Nunc a, lacinia sed risus neque, arcu, rhoncus. Id mauris justo facilisis aliquam platea vestibulum condimentum morbi.",
      icon: <ViewList />,
    },
    {
      title: "Benefit No.2",
      content:
        "Purus lobortis volutpat posuere id integer nunc tellus. Non mauris malesuada feugiat massa mi pellentesque cum est. Pharetra a varius urna rhoncus, tempor rutrum.",
      icon: <ViewBoard />,
    },
    {
      title: "Benefit No.3",
      content:
        "Purus lobortis volutpat posuere id integer nunc tellus. Non mauris malesuada feugiat massa mi pellentesque cum est. Pharetra a varius urna rhoncus, tempor rutrum.",
      icon: <Calendar />,
    },
    {
      title: "Benefit No.4",
      content:
        "Tincidunt sollicitudin interdum nunc sit risus at bibendum vitae. Urna, quam ut sit justo non, consectetur et varius.",
      icon: <Users />,
    },
  ];
  const { container } = render(
    <AthleteBenefits
      title="athletesâ€™ benefits"
      description="Get yourself an opportunity to earn money and interact with your fans at the same time!"
      features={AthleteBenefitsMockData}
    />
  );
  expect(container).toMatchSnapshot();
});
