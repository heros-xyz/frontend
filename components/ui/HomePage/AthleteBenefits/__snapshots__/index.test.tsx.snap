// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders athlete benefits unchanged 1`] = `
<div>
  <div
    class="css-1ey1mgp"
  >
    <div
      class="css-1upilqn"
    >
      <p
        class="chakra-text css-1g26l3f"
      >
        athletesâ€™ benefits
      </p>
      <p
        class="chakra-text css-1pwlws1"
      >
        Get yourself an opportunity to earn money and interact with your fans at the same time!
      </p>
    </div>
    <div
      class="css-1upilqn"
    >
      <div
        class="css-1qu44at"
      >
        <div
          class="css-150lb0i"
        >
          <div
            class="css-k6qc3w"
          >
            <div
              class="css-9uy14h"
            >
              <svg
                class="chakra-icon css-1ewd8qv"
                focusable="false"
                viewBox="0 0 18 14"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M1 1H17M1 5H17M1 9H17M1 13H17"
                  stroke="#313F4C"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                />
              </svg>
            </div>
          </div>
        </div>
        <div
          class="css-18jgh7v"
        >
          <p
            class="chakra-text css-sx37wz"
          >
            Benefit No.1
          </p>
          <p
            class="chakra-text css-1oste6b"
          >
            Nunc a, lacinia sed risus neque, arcu, rhoncus. Id mauris justo facilisis aliquam platea vestibulum condimentum morbi.
          </p>
        </div>
      </div>
      <div
        class="css-1qu44at"
      >
        <div
          class="css-150lb0i"
        >
          <div
            class="css-k6qc3w"
          >
            <div
              class="css-9uy14h"
            >
              <svg
                class="chakra-icon css-1ewd8qv"
                focusable="false"
                viewBox="0 0 20 16"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M7 13V3M7 13C7 14.1046 6.10457 15 5 15H3C1.89543 15 1 14.1046 1 13V3C1 1.89543 1.89543 1 3 1H5C6.10457 1 7 1.89543 7 3M7 13C7 14.1046 7.89543 15 9 15H11C12.1046 15 13 14.1046 13 13M7 3C7 1.89543 7.89543 1 9 1H11C12.1046 1 13 1.89543 13 3M13 13V3M13 13C13 14.1046 13.8954 15 15 15H17C18.1046 15 19 14.1046 19 13V3C19 1.89543 18.1046 1 17 1H15C13.8954 1 13 1.89543 13 3"
                  stroke="#313F4C"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                />
              </svg>
            </div>
          </div>
        </div>
        <div
          class="css-18jgh7v"
        >
          <p
            class="chakra-text css-sx37wz"
          >
            Benefit No.2
          </p>
          <p
            class="chakra-text css-1oste6b"
          >
            Purus lobortis volutpat posuere id integer nunc tellus. Non mauris malesuada feugiat massa mi pellentesque cum est. Pharetra a varius urna rhoncus, tempor rutrum.
          </p>
        </div>
      </div>
      <div
        class="css-1qu44at"
      >
        <div
          class="css-150lb0i"
        >
          <div
            class="css-k6qc3w"
          >
            <div
              class="css-9uy14h"
            >
              <svg
                class="chakra-icon css-nyt2fo"
                focusable="false"
                viewBox="0 0 20 20"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M6 5V1M14 5V1M5 9H15M3 19H17C18.1046 19 19 18.1046 19 17V5C19 3.89543 18.1046 3 17 3H3C1.89543 3 1 3.89543 1 5V17C1 18.1046 1.89543 19 3 19Z"
                  stroke="#313F4C"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                />
              </svg>
            </div>
          </div>
        </div>
        <div
          class="css-18jgh7v"
        >
          <p
            class="chakra-text css-sx37wz"
          >
            Benefit No.3
          </p>
          <p
            class="chakra-text css-1oste6b"
          >
            Purus lobortis volutpat posuere id integer nunc tellus. Non mauris malesuada feugiat massa mi pellentesque cum est. Pharetra a varius urna rhoncus, tempor rutrum.
          </p>
        </div>
      </div>
      <div
        class="css-1qu44at"
      >
        <div
          class="css-150lb0i"
        >
          <div
            class="css-k6qc3w"
          >
            <div
              class="css-9uy14h"
            >
              <svg
                class="chakra-icon css-1ewd8qv"
                focusable="false"
                viewBox="0 0 20 20"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M10 2.35418C10.7329 1.52375 11.8053 1 13 1C15.2091 1 17 2.79086 17 5C17 7.20914 15.2091 9 13 9C11.8053 9 10.7329 8.47624 10 7.64582M13 19H1V18C1 14.6863 3.68629 12 7 12C10.3137 12 13 14.6863 13 18V19ZM13 19H19V18C19 14.6863 16.3137 12 13 12C11.9071 12 10.8825 12.2922 10 12.8027M11 5C11 7.20914 9.20914 9 7 9C4.79086 9 3 7.20914 3 5C3 2.79086 4.79086 1 7 1C9.20914 1 11 2.79086 11 5Z"
                  stroke="#313F4C"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                />
              </svg>
            </div>
          </div>
        </div>
        <div
          class="css-18jgh7v"
        >
          <p
            class="chakra-text css-sx37wz"
          >
            Benefit No.4
          </p>
          <p
            class="chakra-text css-1oste6b"
          >
            Tincidunt sollicitudin interdum nunc sit risus at bibendum vitae. Urna, quam ut sit justo non, consectetur et varius.
          </p>
        </div>
      </div>
    </div>
    <div
      class="css-1jxkzrp"
    >
      <button
        class="chakra-button css-ybm1h3"
        type="button"
      >
        become an athlete
      </button>
    </div>
  </div>
</div>
`;
